homeassistant/components/nederlandse_spoorwegen/sensor.py

```diff
@@ -128,40 +124,75 @@ def device_state_attributes(self):
             for k in self._trips[0].trip_parts:
                 route.append(k.destination)
 
-        return {
+        # Static attributes
+        attributes = {
             "going": self._trips[0].going,
             "departure_time_planned": self._trips[0].departure_time_planned.strftime(
                 "%H:%M"
             ),
-            "departure_time_actual": self._trips[0].departure_time_actual.strftime(
-                "%H:%M"
-            ),
-            "departure_delay": self._trips[0].departure_time_planned
-            != self._trips[0].departure_time_actual,
-            "departure_platform": self._trips[0].trip_parts[0].stops[0].platform,
-            "departure_platform_changed": self._trips[0]
-            .trip_parts[0]
-            .stops[0]
-            .platform_changed,
+            "departure_time_actual": None,
+            "departure_delay": None,
+            "departure_platform_planned": self._trips[0].departure_platform_planned,
+            "departure_platform_actual": None,
             "arrival_time_planned": self._trips[0].arrival_time_planned.strftime(
                 "%H:%M"
             ),
-            "arrival_time_actual": self._trips[0].arrival_time_actual.strftime("%H:%M"),
-            "arrival_delay": self._trips[0].arrival_time_planned
-            != self._trips[0].arrival_time_actual,
-            "arrival_platform": self._trips[0].trip_parts[0].stops[-1].platform,
-            "arrival_platform_changed": self._trips[0]
-            .trip_parts[0]
-            .stops[-1]
-            .platform_changed,
-            "next": self._trips[1].departure_time_actual.strftime("%H:%M"),
+            "arrival_time_actual": None,
+            "arrival_delay": None,
+            "arrival_platform_platform": self._trips[0].arrival_platform_planned,
+            "arrival_platform_actual": self._trips[0].arrival_platform_actual,
+            "direction": self._trips[0].destination,
+            "next": None,
             "status": self._trips[0].status.lower(),
             "transfers": self._trips[0].nr_transfers,
             "route": route,
-            "remarks": [r.message for r in self._trips[0].trip_remarks],
             ATTR_ATTRIBUTION: ATTRIBUTION,
         }
 
+        # Departure attributes
+        if self._trips[0].departure_time_actual is None:
+            attributes["departure_time_actual"] = self._trips[
+                0
+            ].departure_time_planned.strftime("%H:%M")
```

Comment:
This made most sense to me, mainly 2 reasons:
- it is named "actual" so we need to get the actual departure time, this could either be the planned time or a delayed time but in all cases should the "actual" time be the time the train will depart;
- this is a conditional statement that needs to be made at some point. If we set "actual" to None, later you must first check if "actual" is none, if so take "planned", if not take "actual".

I would prefer to not go back entirely to the old way so I propose two solutions:
- we keep two departure times (requiring a conditional later on) but then "actual" needs to be refactored because that name is misleading, or
- we switch to a single departure time, there's already a "delayed" Boolean so that gives you most of the info needed.

Or third solution, we keep this. "Planned" is a reference time, "actual" is the actual time whether planned or delayed.

Reply:
Fixes and enhancements should not go into the same PR. Please limit the attribute changes to those required for the API update (if any).

It is fine to submit further PRs with the remaining proposals.

Reply:
Will do, thanks.